src/main/java/org/cru/godtools/api/v2/PhraseResource.java

```diff
@@ -0,0 +1,231 @@
+
+		return Response
+				.ok(translationElements)
+				.build();
+	}
+
+	@POST
+	@Consumes(MediaType.APPLICATION_JSON)
+	public Response addPhraseToPackage(@HeaderParam("Authorization") String authorization,
+									   @PathParam("package") String packageCode,
+									   @PathParam("pageName") String pageName,
+									   TranslationElement translationElement)
+	{
+		logger.info(String.format("Adding phrase w/ authorization %s", authorization));
+
+		AuthorizationRecord.checkAdminAccess(authorizationService.getAuthorizationRecord(null, authorization), clock.currentDateTime());
+
+		logger.info(String.format("Adding phrase, admin validated.  Adding phrase for %s-%s", packageCode, pageName));
+
+		Optional<GodToolsTranslation> translationOptional = getGodToolsTranslationOptional(packageCode);
+
+		Optional<PageStructure> pageOptional = translationOptional.get().getPage(pageName);
+
+		if(!pageOptional.isPresent())
+		{
+			logger.warn(String.format("Adding phrases, Page %s not found English translation for %s", pageName, packageCode));
+
+			return Response.status(Response.Status.NOT_FOUND).build();
+		}
+
+		Package gtPackage = packageService.selectByCode(packageCode);  // can't be null by this point
+
+		translationElement.setFieldsForNewPhrase(pageOptional.get());
+
+		// TODO: display reordering - minor, perhaps
+
+		for(Language language : languageService.selectAllLanguages())
+		{
+			Translation translation = translationService.selectByLanguageIdPackageIdVersionNumber(language.getId(),
+					gtPackage.getId(),
+					GodToolsVersion.LATEST_VERSION);
+
+			// no translation in this language
+			if(translation == null) continue;
+
+			// i probably don't need to copy here, but it makes me feel better to do it.  in the JPA world it would be
+			// necessary
+			TranslationElement translationElementForOtherLanguage = TranslationElement.copyOf(translationElement);
+
+			translationElementForOtherLanguage.setTranslationId(translation.getId());
+
+			translationElementService.insert(translationElementForOtherLanguage);
+		}
+
+		translationUpload.doUpload(gtPackage.getTranslationProjectId(),
+				BASE_LANGUAGE_CODE,
+				pageName);
+
+		return Response
+				.status(Response.Status.CREATED)
+				.build();
+	}
+
+	@DELETE
+	@Path("/{id}")
+	public Response removePhraseFromPackage(@HeaderParam("Authorization") String authorization,
+											@PathParam("package") String packageCode,
+											@PathParam("pageName") String pageName,
+											@PathParam("id") UUID phraseId)
+	{
+		logger.info(String.format("Removing phrase w/ authorization %s", authorization));
+
+		AuthorizationRecord.checkAdminAccess(authorizationService.getAuthorizationRecord(null, authorization), clock.currentDateTime());
+
+		logger.info(String.format("Removing phrase, admin validated.  Adding phrase for %s-%s", packageCode, pageName));
+
+		Optional<GodToolsTranslation> translationOptional = draftTranslation.retrieve(BASE_LANGUAGE_CODE, packageCode, false);
```

Comment:
Cmd+Opt+m when you have a section of code highlighted will refactor it out into a method.  Then intelliJ will scan that file and see if other portions of that code can also use the extracted method.  It missed this case I think b/c I forgot to switch the return NotFound response to throw NotFoundException.

Like a lot of intelliJ shortcuts, Cmd+Opt+m is really helpful.  Cmd+Opt+v is another must have.. extracts code to a variable